### PR TITLE
Add RIN thresholds for the Rnafusion delivery report

### PR DESCRIPTION
**Pull request by ivadym**
## Description

Documentation: https://github.com/Clinical-Genomics/atlas/pull/2693

### Added
- RIN max and min thresholds for delivery report validation

### Fixed
- Zero integers treated as `N/A` when parsing metrics for the delivery report

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [x] `cg generate delivery-report <case-id>`

### Expected test outcome

- [ ] Report renders N/A for RIN values outside range, including 0 
  - No available cases in LIMS. Tested only locally.
- [x] Report render N/A for empty RIN values.
- [x] Report generation Taxprofiler.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/models/report/validators.py [hunk 1]
```diff
@@ -33,14 +33,18 @@ def get_boolean_as_string(value: bool | None) -> str:
...
 
 
-def get_float_as_string(value: float | None) -> str:
-    """Return string representation of a float value."""
-    return str(round(float(value), PRECISION)) if value or isinstance(value, float) else NA_FIELD
+def get_number_as_string(value: int | float | None) -> str:
+    """Return string representation of a number."""
+    return (
+        str(round(float(value), PRECISION))
+        if value or isinstance(value, (int, float))
```
Comment (diitaz93) [inline cg/models/report/validators.py:40]: Im wondering what would happen if the input is bool (True) or str, would the behaviour be unexpected?
```suggestion
        if isinstance(value, (int, float))
```

**Review by diitaz93 — COMMENTED** [review]
will finish later

File: cg/models/report/validators.py (hunk 1 above)
Comment (ivadym) [reply cg/models/report/validators.py:40]: If bool or an invalid string (not coercible to a float) are passed we would like this to fail actually.

Comment (ivadym) [reply cg/models/report/validators.py]: just edited the comment as I realised some coverage values are parsed as stirngs for mip :( 

>  I will keep the value check as some fields may be parsed as strings

File: tests/meta/report/test_rnafusion_api.py [hunk 2]
```diff
@@ -33,3 +37,35 @@ def test_get_sample_metadata(
 
     # THEN the sample metadata should be correctly retrieved and match the expected validated metrics
     assert sample_metadata.model_dump() == rnafusion_validated_metrics
+
+
+def test_ensure_rin_thresholds(
```
Comment (diitaz93) [inline tests/meta/report/test_rnafusion_api.py:42]: I think more tests with different values of RIN are important, maybe you can parametrise this test

File: tests/models/report/test_validators.py [hunk 3]
```diff
@@ -63,27 +63,53 @@ def test_get_boolean_as_string():
...
 
     # WHEN performing the validation
-    validated_float_value: str = get_float_as_string(float_value)
+    validated_float_value: str = get_number_as_string(float_value)
+
+    # THEN check if the input value was formatted correctly
+    assert validated_float_value == "0.0"
+
+
+def test_get_int_number_as_string():
```
Comment (diitaz93) [inline tests/models/report/test_validators.py:92]: O parametrise this one instead

**Review by diitaz93 — APPROVED** [review]
Looks great 👍 Maybe you can add more tests scenarios to the number to str function

**Comment by ivadym** [comment]
Deployed! Thanks for the review @diitaz93!

**Pull request merged**